sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs

```diff
@@ -174,161 +158,66 @@ internal string OrderByRaw
-                    }
-                }
-
-                return queryCaptionStringValue;
-            }
+            get { return SemanticSearch?.QueryCaption?.QueryCaptionRaw; }
+            set { SemanticSearch.QueryCaption.QueryCaptionRaw = value; }
+        }
 
-            set
-            {
-                if (string.IsNullOrEmpty(value))
-                {
-                    QueryCaption = null;
-                    QueryCaptionHighlightEnabled = null;
-                }
-                else
-                {
-                    int splitIndex = value.IndexOf(QueryCaptionRawSplitter, StringComparison.OrdinalIgnoreCase);
-                    if (splitIndex >= 0)
-                    {
-                        var queryCaptionPart = value.Substring(0, splitIndex);
-                        var highlightPart = value.Substring(splitIndex + QueryCaptionRawSplitter.Length);
+        /// <summary> Allows the user to choose whether a semantic call should fail completely (default / current behavior), or to return partial results. </summary>
+        [CodeGenMember("SemanticErrorHandling")]
+        private SemanticErrorMode? SemanticErrorMode
+        {
+            get { return SemanticSearch?.SemanticErrorMode; }
+            set { SemanticSearch.SemanticErrorMode = value; }
+        }
 
-                        QueryCaption = string.IsNullOrEmpty(queryCaptionPart) ? null : new QueryCaptionType(queryCaptionPart);
-                        QueryCaptionHighlightEnabled = bool.TryParse(highlightPart, out bool highlightValue) ? highlightValue : null;
-                    }
-                    else
-                    {
-                        QueryCaption = new QueryCaptionType(value);
-                        QueryCaptionHighlightEnabled = null;
-                    }
-                }
-            }
+        /// <summary> Allows the user to set an upper bound on the amount of time it takes for semantic enrichment to finish processing before the request fails. </summary>
+        private int? SemanticMaxWaitInMilliseconds
+        {
+            get { return SemanticSearch?.SemanticMaxWaitInMilliseconds?.Milliseconds; }
+            set { SemanticSearch.SemanticMaxWaitInMilliseconds = value.HasValue ? TimeSpan.FromMilliseconds(value.Value) : null; }
         }
 
         /// <summary> The query parameters for multi-vector search queries. </summary>
-        public IList<VectorQuery> VectorQueries { get; internal set; } = new List<VectorQuery>();
+        [CodeGenMember("VectorQueries")]
+        private IList<VectorizableQuery> VectorizableQueries
+        {
+            get { return VectorSearch?.VectorizableQueries; }
+            set { VectorSearch.VectorizableQueries = value; }
+        }
+
+        /// <summary> Determines whether or not filters are applied before or after the vector search is performed. Default is 'preFilter' for new indexes. </summary>
```

Comment:
Nit - if we're customizing this, we should change `'preFilter'` into `<see cref="...VectorFilterMode.PreFilter">`.